apps/store/src/features/carDealership/CarTrialExtensionBlock.tsx

```diff
@@ -87,15 +92,11 @@ const useCarTrialQuery = (params: UseCarTrialQueryParams) => {
     },
   })
 
-  if (data) {
-    return getTrialExtension(data)
-  }
-
-  return null
+  return data
 }
 
 type AddNotificationBannerOptions = {
-  data: TrialExtension
+  data: CarTrialExtension
```

Comment:
Right now `CarTrialExtension` type can assume the "actual data" | null | undefined.
It would be nice if we could right code here assuming `data: CarTrialExtension` is defined and non nullable and make sure we only call `addNotificationBanner` if we actually have some data because right now we need to right:

```
addBanner(
  t('REMAIN_INSURED_BANNER', {
    dueDate: `<b>${dateFull(new Date(data?.trialContract.terminationDate))}</b>`,
  }),
  'warning',
  { force: true },
)
```

and if `data` is nullable here we'll end up showing some weird info in that banner.

But maybe we can do that in another PR, as part of a refactoring.

```suggestion
  data: NonNullable<CarTrialExtension>
```

Reply:
Yes, I had a bit of problems with the the banner and typing :) I was thinking of handling in a new PR but let's sync and we can try to figure it out 🙌 @guilhermespopolin